### ISSUE TITLE
[BUG] Problem with PaCMAP low-dim initialization.

### ISSUE THREAD

**Issue by kenneth-howe-mx** [YingfanWang/PaCMAP#131]
### Bug Description

PaCMAP is very sensitive to the scaling of the initial low-dimensional layout. Given the current scaling approach in `preprocess_X` ([L371-L375](https://github.com/YingfanWang/PaCMAP/blob/3791c0f20b35754c5853d7178f3762be3102f29d/source/pacmap/pacmap.py#L371-L375)), the actual scale of the preprocessed data, and therefore the scale of the resulting PCA solution, is largely determined by the maximum coordinate value in the input data. This is due to dividing by the maximum coordinate value _before_ translation rather than _after_.

This can result in total failure for PaCMAP if the maximum value of the input data is low. See the following example on the 3D mammoth dataset:

```
X = pl.read_csv("./data/mammoth_3d.csv").to_numpy()
X_transformed = PaCMAP(apply_pca=False).fit_transform(X)

Xt = (X.min(axis=0) - X) + 1e-2
Xt_transformed = PaCMAP(apply_pca=False).fit_transform(Xt)
```

PaCMAP embedding of original dataset `X`:

<img width="528" height="413" alt="Image" src="https://github.com/user-attachments/assets/1dc09b8a-6b97-48a8-bce5-ca19700d7299" />

PaCMAP embedding of dataset `Xt` scaled such that maximum value is 0.01:

<img width="555" height="304" alt="Image" src="https://github.com/user-attachments/assets/9a91f2cf-2b14-403e-bd8e-634eb51a4d00" />

### Potential Solution

There are many possible scalings that one can perform here, but here's one that seems to work. Here we translate the minimum of the dataset to the origin and divide by the maximum _after_ translation. (This is equivalent to scaling by the width of the longest axis.) This preserves aspect ratio and is not sensitive to the specific values of the input data:

```
xmin = np.min(X, axis=0)
X -= xmin
xmax = np.max(X)
X /= xmax
xmean = np.mean(X, axis=0)
X -= xmean
```

Here is the result of using PaCMAP with the proposed scaling on the `Xt` example above:

<img width="529" height="413" alt="Image" src="https://github.com/user-attachments/assets/a844efe4-f459-4228-856a-d7388d04bec6" />

This now works as expected.

I would be happy to drop a PR if desired.

**Comment by williamsyy** [comment]
@kenneth-howe-mx Thank you for your interest in PaCMAP. You are right that when the ```xmax``` value is extremely small, then during the preprocessing, it would become large and ruin the whole embedding. The current master branch have fixed this by modifying ```preprocess_X``` to

```python
xmin = np.min(X)
X -= xmin
xmax = np.max(X)
X /= xmax
xmean = np.mean(X, axis=0)
X -= xmean
```

Please refer to bd20dfc4c43b7d4a3d2c49fac5febfb1706b50f2 for more details. Again, thanks for your interested in PaCMAP.
